parser.go

```diff
@@ -576,31 +636,200 @@ func cleanHeredocStart(token lexer.Token) (lexer.Token, error) {
 	return token, nil
 }
 
+// ParseOption represents a functional option for Parse, ParseString, and ParseBytes.
+type ParseOption func(*parseConfig)
+
+// parseConfig holds the configuration for parsing.
+type parseConfig struct {
+	detachedComments bool
+}
+
+// WithDetachedComments controls whether detached comments are preserved in the AST.
+// If set to false (default), Comment nodes will be stripped from the AST during post-processing.
+// If set to true, Comment nodes will be preserved as separate entries in the AST.
```

Comment:
This reads as if all comment nodes will be stripped, which I assume is not what it actually does? I think it would be good to clarify what detached comments are, perhaps:

```suggestion
// WithDetachedComments controls whether comments that are not directly associated with a
// block or attribute are preserved in the AST.
// If set to false (default), detached comments will be stripped from the AST during post-processing.
// If set to true, detached comments will be preserved as separate entries in the AST.
```

Reply:
Thanks! That's better 👍🏼